projects/storefrontlib/src/cms-components/myaccount/update-email/update-email.component.ts

```diff
@@ -1,69 +1,32 @@
-import { Component, OnDestroy, OnInit } from '@angular/core';
-import {
-  AuthService,
-  GlobalMessageService,
-  GlobalMessageType,
-  RoutingService,
-  UserService,
-} from '@spartacus/core';
-import { Observable, Subscription } from 'rxjs';
+import { ChangeDetectionStrategy, Component, OnDestroy, OnInit } from '@angular/core';
+import { Observable } from 'rxjs';
+import { UpdateEmailService } from './update-email.service';
+import { FormGroup } from '@angular/forms';
 
 @Component({
   selector: 'cx-update-email',
   templateUrl: './update-email.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class UpdateEmailComponent implements OnInit, OnDestroy {
   constructor(
-    private routingService: RoutingService,
-    private globalMessageService: GlobalMessageService,
-    private userService: UserService,
-    private authService: AuthService
-  ) {}
-
-  private subscription = new Subscription();
-  private newUid: string;
+    protected updateEmailService: UpdateEmailService
+  ) {
+  }
+  updateEmailForm: FormGroup;
   isLoading$: Observable<boolean>;
```

Comment:
It's actually `isUpdating$`. `isLoading$` indicates something that is done upfront.